### PR TITLE
Move registration of FlushHostnameCache to TenancyServiceProvider::boot 

### PR DESCRIPTION
**Pull request by JaZo**
## Description

I moved the registration of `FlushHostnameCache` to `TenancyServiceProvider::boot`.

## Motivation and context

The event dispatcher, which is required for model events, is only available in the boot method of a service provider. This event dispatcher is setup in `\Illuminate\Database\DatabaseServiceProvider::boot`. Registering an event observer earlier will cause the model to be booted too early when it has no event dispatcher. A model will only boot once, so if the consumer has custom logic in his model boot method that requires an event dispatcher, that functionality is also silently broken (that's how I discovered this issue).

### TL;DR
`FlushHostnameCache` is not activated.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by luceos** [comment]
Thanks for the great clarification. Totally makes sense!
